crates/core/src/kernel/snapshot/replay.rs

```diff
@@ -311,7 +312,7 @@ fn parse_partitions(batch: RecordBatch, partition_schema: &StructType) -> DeltaR
 
     insert_field(
         batch,
-        StructArray::try_new(
+        StructArray::try_new_with_length(
```

Comment:
I updated a few calls to use `StructArray::try_new_with_length(` as without them several of the unit tests were failing